tests/python/gaia-ui-tests/gaiatest/apps/fmradio/app.py

```diff
@@ -81,7 +83,14 @@ def is_power_button_on(self):
 
     @property
     def frequency(self):
-        return float(self.marionette.find_element(*self._frequency_display_locator).text)
+        raw_frequency = self.marionette.find_element(*self._frequency_display_locator).text
+        return float(self._crop_trailing_mhz_and_invisible_characters(raw_frequency))
+
+    @staticmethod
+    def _crop_trailing_mhz_and_invisible_characters(raw_frequency):
+        match = re.search(r'\d{2}\.\d', raw_frequency)
```

Comment:
Hmm, in this case, we always assumes that the FM frequency is in following format: XX.XMhz.  But since FM frequency can go above 3 digits after the decimal point, (eg. "107.5MHz") it'll not work properly in such case.  If we're only going to check the 87.5Mhz and have no need for general method, perhaps we should make it clear in the method name.  Could we remove anything that is non-digit and decimal point instead?  like, this would work:
match = re.search(r'\d+\.\d',raw_frequency)